main.go

```diff
@@ -100,15 +122,25 @@ func findIDEDistrubutionLogsPath(output string) (string, error) {
 	return "", nil
 }
 
-func generateExportOptionsPlist(exportMethodStr, teamID string, uploadBitcode, compileBitcode bool, xcodebuildMajorVersion int64, archive xcarchive.IosArchive) (string, error) {
+func generateExportOptionsPlist(exportProduct ExportProduct, exportMethodStr, teamID string, uploadBitcode, compileBitcode bool, xcodebuildMajorVersion int64, archive xcarchive.IosArchive) (string, error) {
 	log.Printf("Generating export options")
 
+	var productBundleID string
 	var exportMethod exportoptions.Method
 	exportTeamID := ""
 	exportCodeSignIdentity := ""
 	exportProfileMapping := map[string]string{}
 	exportCodeSignStyle := ""
 
+	switch exportProduct {
+	case ExportProductApp:
+		productBundleID = archive.Application.BundleIdentifier()
+		break
+	case ExportProductAppClip:
+		productBundleID = archive.Application.ClipApplication.BundleIdentifier()
```

Comment:
Is the nil handled correctly (archive.Application.ClipApplication) if ExportProductAppClip is selected, but there is no App Clip? Same for archive.Application.

Reply:
Added check

Reply:
Would move this check after `archive, err := xcarchive.NewIosArchive(configs.ArchivePath)` on line 433 of main.go. Calling fail() from a function is not a good practice.

Reply:
done